Get-item-rework-rando-next

leggettc18 opened the pull request:
A PR of my Item Table Rework into rando-next. Playtesting appreciated! Crashes and bugs should probably be expected.

Ice Traps currently have some strange behavior in certain situations, that is a bug but not specific to this build.

leggettc18 committed:
Adds ItemTableManager class.

leggettc18 committed:
Implements new getItem table in game.

leggettc18 committed:
Adds rando item table and way to differentiate tables in GetItemEntry.

leggettc18 committed:
Resolve merge conflicts with develop-zhora

leggettc18 committed:
Adds rough ability to differentiate between rando and vanilla items.merge stashed changes from before develop-zhora merge

leggettc18 committed:
Change ItemTableID to be uint16_t so we can use ModIndex for it.

leggettc18 committed:
Should fix switch build

leggettc18 committed:
Should fix switch build pt 2

leggettc18 committed:
merge in develop-zhora

leggettc18 committed:
Adds new files to CMakeLists.

leggettc18 committed:
Implements fixes for competing getItem calls.

leggettc18 committed:
Correctly renders freestanding items

Particle effects are probably broken, need to fix them still, I pretty
much know how I would do that.

leggettc18 committed:
Fixed Particle effects in the new getItem system.

leggettc18 committed:
Fixes item fanfares

leggettc18 committed:
Partially fixes Ice Traps

Obtaining a freestanding Ice Trap causes link to slide forward
and receiving one from an NPC plays the sound effect and damage
animation but doesn't freeze link.

leggettc18 committed:
Merge branch 'develop-zhora' of https://github.com/HarbourMasters/Shipwright into get-item-rework

leggettc18 committed:
Some more partial ice trap fixing that wasn't pushed earlier

leggettc18 committed:
Merge branch 'develop-zhora' of https://github.com/HarbourMasters/Shipwright into get-item-rework

leggettc18 committed:
Removes unused function override

leggettc18 committed:
Replaces ::find with ::at and adds exception handling

leggettc18 committed:
Removes some commented out code.

leggettc18 committed:
Refactors rando's GetItemEntry array into two arrays.

One array is for the vanilla items that don't have GetItemEntries in
vanilla, the other is for rando exclusive items. They are stored in
separate arrays before getting added to the table so that we can apply
different modIndexes. The items in the first table have are handled
by the vanilla Item_Give, and the second table needed a custom
`Randomizer_Item_Give` function.

leggettc18 committed:
Renames, relocates, and implements ModIndex enum.

leggettc18 committed:
Removes now unused ItemIDs and GetItemIDs

Also makes all the necessary changes to other code that was still
using them indirectly through the GI to GID map that was removed.
There's quite a lot of changes here and I haven't had time to test them
yet.

leggettc18 committed:
Re-implements GIMESSAGE_UNTRANSLATED as macro

leggettc18 committed:
Removes commented out function.

leggettc18 committed:
Throws exception if an invalid itemID is used

Addresses https://github.com/HarbourMasters/Shipwright/pull/1050#discussion_r943694857

leggettc18 committed:
Removes ARRAY_SIZE in favor of ARRAY_COUNT

ARRAY_COUNT already exists in `macros.h`, I just didn't find it before.

Addresses https://github.com/HarbourMasters/Shipwright/pull/1050#discussion_r943153833

leggettc18 committed:
Inverts CheckContainsRandoItem to CheckContainsVanillaItem.

Addresses https://github.com/HarbourMasters/Shipwright/pull/1050#discussion_r940895135

leggettc18 committed:
Cleanup, bugfixes, removing the `- 1`s from `z_player.c`

leggettc18 committed:
Fixes some funky formatting that got committed earlier.

leggettc18 committed:
Adds else if to added fanfare sound cases.

Addresses https://github.com/HarbourMasters/Shipwright/pull/1050#discussion_r940112924 and https://github.com/HarbourMasters/Shipwright/pull/1050#discussion_r940113492

leggettc18 committed:
Extends GetItemEntry to include getItemId

Also adapts some existing calls for both the entry and the id to only
get the entry.

leggettc18 committed:
Extends GetItemEntry to include GID.

This allows for using it later when drawing freestanding items.

Addresses https://github.com/HarbourMasters/Shipwright/pull/1050#discussion_r943168136

leggettc18 committed:
Rando-specific items use new textId again.

This got lost when merging develop-zhora in because I didn't have custom
messages merged when I started this.

leggettc18 committed:
Sets global modIndex to MOD_NONE on scene load

Fixes a crash when buying items in shops due to them
not triggering the rando code that normally sets these items.
May have also been crashing vanilla playthroughs.

leggettc18 committed:
Realized I had the bgm conditions wrong.

leggettc18 committed:
Fixes "static drops" (i.e. sticks from withered babas)

leggettc18 committed:
Fixes LACS/Prelude situation... again.

leggettc18 committed:
Fixes too many arguments error.

Not sure why this didn't fail to build on Windows before.

leggettc18 committed:
Merge branch 'rando-next' of https://github.com/HarbourMasters/Shipwright into get-item-rework-rando-next

Hopefully everything works here. Need to do some testing.

leggettc18 committed:
Fixes Link's Pocket items.

leggettc18 committed:
Simplifies sram init for rando-specific items

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

leggettc18 committed:
Fixes build error for Windows and Mac

leggettc18 commented:
Known Bugs

- [x] Deku Tree top room Skulltula and Chest had the Shadow Medallion. Spoilerlog indicates that the Skulltula was supposed to Shadow Medallion and the chest was supposed to be a Blue Rupee. (Possibly Fixed?)
- [x] Picking up Ice Traps from certain skulltulas results in Link attempting to hold it over his head, crashing the game.
- [x] Approaching bottleable items seems to either trigger get item code or crash the game.
- [x] Some skulltula ice traps don't go through at all.
- [x] There's some weird behavior around bottleable critters where you can't pick up objects while they are nearby. For example in Dodongo's cavern there are two pots, one of which contains a fairy. If you release the fairy you can't pick up the other pot until you collect/bottle the fairy. (Apparently this is actually normal behavior, experienced it outside my branch as well. Might need more investigation).

Sarge-117 commented:
Only thing I can see in my (brief) test so far is that `Audio_PlayFanfare_Rando` was updated to the new IDs in `code_800EC960.c`, but doesn't appear to be getting called anywhere in `z_player.c` anymore.  It is still present in `z_en_si.c`, though, so fanfares won't be consistent between chests and skulls.

leggettc18 committed:
Should fix some Skulltula oddities.

leggettc18 commented:
Ah thanks for the report on this one! I did probably accidentally remove it or circumvent it via if blocks in the merge. Should be a quick fix.

leggettc18 committed:
Cleans up and updates item fanfares.

leggettc18 committed:
Wraps up a few missed type changes.

leggettc18 committed:
Fixes Link's Pocket being a Dungeon Reward from Impa.

leggettc18 committed:
Fixes crashes with `Audio_PlayFanfare_Rando`.

leggettc18 committed:
Some more changes to account for Skulltulas in new system.

leggettc18 committed:
Should fix Skulltula Ice Traps.

aMannus committed:
Eliminated copy pasta

aMannus committed:
Fixed newline end of file

leggettc18 committed:
Merge pull request #1 from aMannus/tokensanity-code-cleanup

Tokensanity code cleanup

leggettc18 committed:
Another Skulltula Ice Trap fix.

leggettc18 committed:
Fixes issues with approaching bottleable items.

leggettc18 committed:
Fixes issues with approaching bottleable items.

leggettc18 committed:
Fixes Ruto's Letter.

It was accidentally getting classified as a rando item.

leggettc18 committed:
Fixes Ruto's Letter.

It was accidentally getting classified as a rando item.

leggettc18 committed:
Should re-fix freestanding ice traps

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

leggettc18 committed:
Prevents Skulltulas from affecting other nearby checks.

leggettc18 committed:
Fixes a few more freestanding items that weren't covered.

Sarge-117 commented:
> Ah thanks for the report on this one! I did probably accidentally remove it or circumvent it via if blocks in the merge. Should be a quick fix.

No problem! Another small issue though - the code in `code_800EC960.c` under the Cvar `gRandoQuestItemFanfares` never gets called, even with the Cvar toggled on.  The rest of the fanfares seem to be working correctly though :)

leggettc18 committed:
Makes freestanding items set player->getItemEntry.

This prevents freestanding items from setting the global modIndex.
This is part of a larger transition that needs to happen to switch
to setting getItemEntries for all of the rando items. This prevents
some things that set getItemId of GI_MAX from granting a Fire Medallion
when the global modIndex is MOD_RANDOMIZER.

leggettc18 committed:
Makes sure we aren't using getItemEntry when not randoed.

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

garrettjoecox committed:
Replace Randomizer_GetRandomizedItemId with Randomizer_GetRandomizedItem and Randomizer_GetItemIdFromKnownCheck with Randomizer_GetItemFromKnownCheck

garrettjoecox committed:
Move check for gRandoQuestItemFanfares into MOD_NONE block within Audio_PlayFanfare_Rando

garrettjoecox committed:
Introduce some new methods and migrate most actors to them

leggettc18 committed:
Merge pull request #3 from garrettjoecox/get-item-rework-rando-next-fanfare-fix

Move check for gRandoQuestItemFanfares into MOD_NONE block within Audio_PlayFanfare_Rando

leggettc18 committed:
Fixes ocarina game skull kids to set player->getItemEntry

leggettc18 committed:
Sets `z_en_box.c` to set `player->getItemEntry`

garrettjoecox committed:
Fix logical errors and migrate most of the rest of the rando checks to GiveItemEntryFromActor

leggettc18 committed:
Merge branch 'get-item-rework-help1' of github.com:garrettjoecox/Shipwright into pr/garrettjoecox/2

garrettjoecox committed:
Use GiveItemEntryFromActorWithFixedRange in item00

leggettc18 committed:
Fixes Anju to set player->getItemEntry.

leggettc18 committed:
Merge branch 'get-item-rework-help1' of github.com:garrettjoecox/Shipwright into pr/garrettjoecox/2

garrettjoecox committed:
Add a few missing cases

leggettc18 committed:
Merge branch 'get-item-rework-help1' of github.com:garrettjoecox/Shipwright into pr/garrettjoecox/2

leggettc18 committed:
Merge pull request #2 from garrettjoecox/get-item-rework-help1

Replace Randomizer_GetRandomizedItemId with Randomizer_GetRandomizedI…

leggettc18 committed:
Additional fix for Skull Kid

leggettc18 committed:
Merge branch 'get-item-rework' of github.com:leggettc18/Shipwright into get-item-rework

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

leggettc18 committed:
Merge get-item-rework and update new checks.

leggettc18 committed:
Fixes chests by allowing for negative getItemIds

leggettc18 committed:
Fixes Zora's River Frog checks.

leggettc18 committed:
Fixes OoT rendering and Ice Trap checks.

leggettc18 committed:
Fixes some Sheik checks I had accidentally set the wrong flag for.

GaryOderNichts committed:
Fix Wii U stick input

leggettc18 committed:
Fixes fanfare getItemId overlap

leggettc18 committed:
Merge branch 'rando-next' of https://github.com/HarbourMasters/Shipwright into get-item-rework-rando-next

Kenix3 committed:
Merge pull request #1241 from GaryOderNichts/fix/wiiu-stick-input

Fix Wii U stick input

qurious-pixel committed:
MacOS OTR (#973)

leggettc18 committed:
Merge branch 'rando-next' of https://github.com/HarbourMasters/Shipwright into get-item-rework-rando-next

leggettc18 committed:
Updates Zora's Sapphire Cutscene to use GetItemEntry

briaguya0 committed:
only toggle menu with controller when controller nav enabled (#1043)

leggettc18 committed:
Merge branch 'develop-zhora' of github.com:HarbourMasters/Shipwright into get-item-rework

leggettc18 committed:
Fixes vanilla ice traps and randomized ice smoke

briaguya0 committed:
Fix controller nav on WiiU (#1258)

Co-authored-by: briaguya <briaguya@alice>

leggettc18 committed:
Merge branch 'rando-next' of github.com:HarbourMasters/Shipwright into get-item-rework-rando-next

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

leggettc18 committed:
Fixes ice traps again.

leggettc18 commented:
This has been pretty thoroughly playtested in both vanilla and randomizer modes and we haven't found any further issues with this or the develop-zhora version.

garrettjoecox reviewed (approved):
Let’s get it in!

leggettc18 committed:
Fixes rendering of treasure chest game items.

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

leggettc18 committed:
Removes unused `Randomizer_GetItemIdFromGetItemId`.

leggettc18 committed:
Cleans up an if statement for item00.

leggettc18 committed:
Cleans up another if statement in item00

leggettc18 committed:
This should fix a bug with the Gerudo Archery minigame.

I wasn't able to get the bug to happen after making this change.

leggettc18 committed:
Merge remote-tracking branch 'origin' into get-item-rework

leggettc18 committed:
Merge branch 'rando-next', remote-tracking branch 'origin' into get-item-rework-rando-next

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

leggettc18 committed:
Fixes goofup of ImGui window from bad merge conflict resolution.

leggettc18 committed:
Merge branch 'get-item-rework' into merge-fix

leggettc18 committed:
merge in get-item-rework-rando-next

leggettc18 committed:
Documents our new GiveItemEntry fuctions.

leggettc18 committed:
Uses more descriptive type name for ItemIDs for creating custom messages.

leggettc18 committed:
Fixes potential issue with if statement.

leggettc18 committed:
Merge branch 'get-item-rework' into get-item-rework-rando-next

leggettc18 committed:
Fixes missed type change.
